libs/utils/src/fs_ext.rs

```diff
@@ -15,6 +15,40 @@ where
     }
 }
 
+///
+/// Command with ability not to give all file descriptors to child process
+///
+pub trait CloseFileDescriptors: CommandExt {
+    ///
+    /// Close file descriptors (other than stdin, stdout, stderr) in child process
+    ///
+    fn close_fds(&mut self) -> &mut Command;
+}
+
+impl<C: CommandExt> CloseFileDescriptors for C {
+    fn close_fds(&mut self) -> &mut Command {
+        unsafe {
+            self.pre_exec(move || {
+                // SAFETY: Code executed inside pre_exec should have async-signal-safety,
+                // which means it should be safe to execute inside a signal handler.
+                // The precise meaning depends on platform. See `man signal-safety`
+                // for the linux definition.
+                //
+                // The set_fds_cloexec_threadsafe function is documented to be
+                // async-signal-safe.
+                //
+                // Aside from this function, the rest of the code is re-entrant and
+                // doesn't make any syscalls. We're just passing constants.
+                //
+                // NOTE: It's easy to indirectly cause a malloc or lock a mutex,
+                // which is not async-signal-safe. Be careful.
+                close_fds::set_fds_cloexec_threadsafe(3, &[]);
+                Ok(())
+            })
+        }
+    }
+}
+
```

Comment:
Placing this in `fs_ext.rs` instead of `process_ext.rs` feels a bit off, as this doesn't have anything to do with filesystem, but you probably considered this already and I assume we don't have any other `CommandExt`.

Reply:
So far, decided to avoid creating new modules and use a small trait that does an fs-related thing in the fs module.

Reply:
Yeah, this can be done later. I will probably make a version of this for tokio, and a version which handles an array of file descriptors, I can split it up then.